presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ProjectOffPushDownRule.java

```diff
@@ -56,13 +56,15 @@ protected ProjectOffPushDownRule(Pattern<N> targetPattern)
     }
 
     @Override
-    public Optional<PlanNode> apply(ProjectNode parent, Captures captures, Context context)
+    public Result apply(ProjectNode parent, Captures captures, Context context)
     {
         N targetNode = captures.get(targetCapture);
 
         return pruneInputs(targetNode.getOutputSymbols(), parent.getAssignments().getExpressions())
                 .flatMap(prunedOutputs -> this.pushDownProjectOff(context.getIdAllocator(), targetNode, prunedOutputs))
-                .map(newChild -> parent.replaceChildren(ImmutableList.of(newChild)));
+                .map(newChild -> parent.replaceChildren(ImmutableList.of(newChild)))
+                .map(Result::ofPlanNode)
+                .orElse(Result.empty());
```

Comment:
If this patter starts to repeat, we might think about simplifying it somehow. I counted 5 occurrences within this PR. (don't have suggestions though)